adapterearthengine/middleware.py

```diff
@@ -30,21 +32,31 @@ def wrapper(*args, **kwargs):
         logging.debug("Getting dataset from id")
 
         try:
-            dataset_object = DatasetService.get(kwargs['dataset_id'])
+            dataset_object = DatasetService.get(
+                kwargs["dataset_id"], request.headers.get("x-api-key")
+            )
         except DatasetNotFound:
-            return error(status=404, detail="Dataset with id {} doesn't exist".format(kwargs['dataset_id']))
+            return error(
+                status=404,
+                detail="Dataset with id {} doesn't exist".format(kwargs["dataset_id"]),
```

Comment:
Just a suggestion, but I would replace this with an f-string, personally. So:
f"Dataset with id {kwargs['dataset_id']} doesn't exist"

Reply:
Then again, I just realized that the only reason this was touched was it was probably modified by an automatic linter, so nevermind. :)